packages/discovery-provider/src/tasks/entity_manager/entities/event.py

```diff
@@ -0,0 +1,113 @@
+from src.exceptions import IndexingValidationError
+from src.models.events.event import Event
+from src.tasks.entity_manager.utils import (
+    EntityType,
+    ManageEntityParameters,
+    copy_record,
+    validate_signer,
+)
+from src.utils.structured_logger import StructuredLogger
+
+logger = StructuredLogger(__name__)
+
+
+def validate_create_event_tx(params: ManageEntityParameters):
+    validate_signer(params)
+    metadata = params.metadata
+
+    required_fields = ["event_type"]
+    for field in required_fields:
```

Comment:
I think probably a few more validation items needed here?
* Event with id doesn't already exist
* Validation for the various rules for remix competition types (has an entity_id and entity_type, no existing remix competition for that entity_id, etc)
* Can't be added in a delete state? (Or we just ignore that value when indexing)
* end_date has to be in the future.
* If entity_id, user_id, those records have to exist. (Which means also making sure they are fetched when we do the related entities fetch)